DEVGUIDE.md

```diff
@@ -275,162 +275,32 @@ Where `<version>` corresponds to the latest Visual Studio version on your machin
-
-     ```shell
-     dotnet add reference ..\..\..\src\FSharp.Core\FSharp.Core.fsproj
-     ```
-
-    > as well as the following property have to be added to `FcsBench.fsproj`:
-
-    ```xml
-    <PropertyGroup>
-        <DisableImplicitFSharpCoreReference>true</DisableImplicitFSharpCoreReference>
-    </PropertyGroup>
-    ```
-
-5. Add a new benchmark for FCS/FSharp.Core by editing `Program.fs`.
-
-      ```fsharp
-      open System.IO
-      open FSharp.Compiler.CodeAnalysis
-      open FSharp.Compiler.Diagnostics
-      open FSharp.Compiler.Text
-      open BenchmarkDotNet.Attributes
-      open BenchmarkDotNet.Running
-
-      [<MemoryDiagnoser>]
-      type CompilerService() =
-          let mutable checkerOpt = None
-          let mutable sourceOpt = None
-
-          let parsingOptions =
-              {
-                  SourceFiles = [|"CheckExpressions.fs"|]
-                  ConditionalDefines = []
-                  DiagnosticOptions = FSharpDiagnosticOptions.Default
-                  LangVersionText = "default"
-                  IsInteractive = false
-                  LightSyntax = None
-                  CompilingFsLib = false
-                  IsExe = false
-              }
-
-          [<GlobalSetup>]
-          member _.Setup() =
-              match checkerOpt with
-              | None ->
-                  checkerOpt <- Some(FSharpChecker.Create(projectCacheSize = 200))
-              | _ -> ()
-
-              match sourceOpt with
-              | None ->
-                  sourceOpt <- Some <| SourceText.ofString(File.ReadAllText("""C:\Users\vlza\code\fsharp\src\Compiler\Checking\CheckExpressions.fs"""))
-              | _ -> ()
+To exercise the benchmarking infrastructure locally, run:
 
+(Windows)
+```cmd
+build.cmd -configuration Release -testBenchmarks
+```
 
-          [<Benchmark>]
-          member _.ParsingTypeCheckerFs() =
-              match checkerOpt, sourceOpt with
-              | None, _ -> failwith "no checker"
-              | _, None -> failwith "no source"
-              | Some(checker), Some(source) ->
-                  let results = checker.ParseFile("CheckExpressions.fs",  source, parsingOptions) |> Async.RunSynchronously
-                  if results.ParseHadErrors then failwithf "parse had errors: %A" results.Diagnostics
-
-          [<IterationCleanup(Target = "ParsingTypeCheckerFs")>]
-          member _.ParsingTypeCheckerFsSetup() =
-              match checkerOpt with
-              | None -> failwith "no checker"
-              | Some(checker) ->
-                  checker.InvalidateAll()
-                  checker.ClearLanguageServiceRootCachesAndCollectAndFinalizeAllTransients()
-                  checker.ParseFile("dummy.fs", SourceText.ofString "dummy", parsingOptions) |> Async.RunSynchronously |> ignore
+(Linux/Mac)
+```shell
+./build.sh -configuration Release -testBenchmarks
```

Comment:
This won't work:
<img width="884" alt="image" src="https://github.com/dotnet/fsharp/assets/1260985/379124c1-c7f2-48d0-b9bf-f29ebe5a868f">
Even if fixed to `./build.sh --configuration Release --testBenchmarks`, it'll still fail
<img width="375" alt="image" src="https://github.com/dotnet/fsharp/assets/1260985/aacaf4cc-6036-4f2e-b73d-4188ab48bd40">

CLI switch needs to be added to script.

Reply:
Fixed